cobalt/build/gn.py

```diff
@@ -90,11 +92,26 @@ def main(out_directory: str, platform: str, build_type: str,
       default=False,
       action='store_true',
       help='Pass this flag to disable the header dependency gn check.')
+  parser.add_argument(
+      '--modular',
+      default=None,
+      action='store_true',
+      help='Pass this flag to build linux modularly.')
+
   script_args, gen_args = parser.parse_known_args()
 
   if not script_args.no_check:
     gen_args.append('--check')
 
+  # --modular flag is added as a convenience for devs to build
+  # modularly on linux. Note that linux crosstool builds dont
+  # build modularly.
+  if script_args.modular:
+    modular_flag_set_correctly = 'linux' in script_args.platform \
+    and not 'crosstool' in script_args.platform
+    assert modular_flag_set_correctly, \
+      'Modular flag should be set only for linux platforms'
+
   if script_args.out_directory:
```

Comment:
Consider modifying output directory here with a `-modular` suffix. It's a matter of preference probably, but at least in most all my uses i want to have monolithic and modular builds side by side in the default output directory, rather than one overwrite the other.

E.g. for the default directory picking code:
```
modular_suffix = ''
if script_args.modular:
  modular_suffix = '-modular'
  
builds_out_directory = os.path.join(
        builds_directory, f'{script_args.platform}{modular_suffix}_{script_args.build_type}')
```